### PR TITLE
Added RDS support

### PR DESCRIPTION
**Pull request by nikitashalabaev**
GET: https://sdyaa5isjk.execute-api.eu-west-1.amazonaws.com/dev/products
GET: https://sdyaa5isjk.execute-api.eu-west-1.amazonaws.com/dev/products/{productId}
POST: https://sdyaa5isjk.execute-api.eu-west-1.amazonaws.com/dev/products

FE: https://d16vxnfhjf4opx.cloudfront.net/

### CODE REVIEW

**Comment by MaxPlav** [comment]
Looks like Frontend app https://d1fhg8k1jichva.cloudfront.net/ is not integrated still

**Comment by MaxPlav** [comment]
https://sdyaa5isjk.execute-api.eu-west-1.amazonaws.com/dev/products?productId=a72f551b-fc49-4802-8b44-3b7433d76d5d should return 1 product but not a list

File: products-service/src/functions/getProductsById/handler.ts [hunk 1]
```diff
@@ -1,19 +1,31 @@
...
 
 const getProductsById = async (event) => {
+  if (!pool) {
+    pool = new Pool(dbConfig);
+  }
+
+  const client = await pool.connect();
+
   try {
     const { productId } = event.pathParameters;
```
Comment (MaxPlav) [inline products-service/src/functions/getProductsById/handler.ts:18]: you should take productId from `queryStringParameters`, as your url is 
`https://sdyaa5isjk.execute-api.eu-west-1.amazonaws.com/dev/products?productId=someId`
if you want to use `pathParameters `, then the url should be
`https://sdyaa5isjk.execute-api.eu-west-1.amazonaws.com/dev/products/someId`

File: products-service/src/functions/seedProductsService/handler.ts [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+      (
+        id UUID PRIMARY KEY DEFAULT uuid_generate_v4(),
+        title VARCHAR NOT NULL,
+        description VARCHAR,
+        price INTEGER
+      );
+      
+      CREATE TABLE IF NOT EXISTS stocks
+      (
+        product_id UUID REFERENCES products (id),
```
Comment (MaxPlav) [inline products-service/src/functions/seedProductsService/handler.ts:29]: you'd better add some 'id' as a primary key

File: products-service/src/functions/createProduct/handler.ts [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+    pool = new Pool(dbConfig);
+  }
+
+  const client = await pool.connect();
+
+  try {
+    const { title, description, price } = event.body;
+
+    await client.query(`
+      INSERT INTO products (title, description, price) VALUES ('${title}', '${description}', ${Number(price)})
```
Comment (MaxPlav) [inline products-service/src/functions/createProduct/handler.ts]: what about insert 'count' to 'stocks' table accordingly?

File: products-service/src/functions/getProductsById/handler.ts (hunk 1 above)
Comment (nikitashalabaev) [reply products-service/src/functions/getProductsById/handler.ts:18]: > you should take productId from `queryStringParameters`, as your url is
> `https://sdyaa5isjk.execute-api.eu-west-1.amazonaws.com/dev/products?productId=someId`
> if you want to use `pathParameters `, then the url should be
> `https://sdyaa5isjk.execute-api.eu-west-1.amazonaws.com/dev/products/someId`

Sorry, wrong link in the description. Updated.

File: products-service/src/functions/seedProductsService/handler.ts (hunk 2 above)
Comment (nikitashalabaev) [reply products-service/src/functions/seedProductsService/handler.ts:29]: Had the same thoughts. But decided to leave structure according to the task description.

**Comment by nikitashalabaev** [comment]
> Looks like Frontend app https://d1fhg8k1jichva.cloudfront.net/ is not integrated still

Deployed updated FE.

**Comment by musiichuk** [comment]
Do you have a transactions here?

**Comment by nikitashalabaev** [comment]
> Do you have a transactions here?

Implementation is without transactions.

**Comment by wwrcth** [comment]
@nikitashalabaev As I can see, you still hasn't integrate fe part with your be.
I can't see a call to products endpoint from your fe app
image

File: products-service/serverless.ts [hunk 4]
```diff
@@ -23,11 +25,16 @@ const serverlessConfiguration: AWS = {
     },
     environment: {
       AWS_NODEJS_CONNECTION_REUSE_ENABLED: '1',
+      PG_HOST: '',
```
Comment (wwrcth) [inline products-service/serverless.ts:28]: Don't you want to setup it here too?

File: products-service/src/functions/getProductsById/handler.ts [hunk 5]
```diff
@@ -1,19 +1,31 @@
...
 
-    const product = await ProductsRepo.getProductsById(productId);
+    const { rows: [product] } = await client.query(`
+      SELECT id, title, description, price, count FROM products p LEFT JOIN stocks s ON p.id = s.product_id WHERE id='${productId}' 
+    `);
 
     return formatJSONResponse({ response: product });
   } catch (e) {
```
Comment (wwrcth) [inline products-service/src/functions/getProductsById/handler.ts:27]: Do you support 404 case? It should be done according to requirements from the 3rd task.

**Comment by nikitashalabaev** [comment]
> @nikitashalabaev As I can see, you still hasn't integrate fe part with your be.
> I can't see a call to products endpoint from your fe app
> image

The correct link to FE: https://d16vxnfhjf4opx.cloudfront.net/ (Updated description)

File: products-service/serverless.ts (hunk 4 above)
Comment (nikitashalabaev) [reply products-service/serverless.ts:28]: Not sure what you've meant.

File: products-service/src/functions/getProductsById/handler.ts (hunk 5 above)
Comment (nikitashalabaev) [reply products-service/src/functions/getProductsById/handler.ts:27]: No 404 support, maybe  i missed something, could you point that requirement?

Comment (wwrcth) [reply products-service/src/functions/getProductsById/handler.ts:27]: Sure, task 3 https://github.com/rolling-scopes-school/nodejs-aws-tasks/blob/main/task3-product-api/task.md
Main error scenarious are handled by API ("**Product not found**" error, try catch blocks are used in lambda handlers).

File: products-service/serverless.ts (hunk 4 above)
Comment (wwrcth) [reply products-service/serverless.ts:28]: Just remember from our example from lesson that db connection config is additionally setup in serverless file too

File: products-service/src/functions/getProductsById/handler.ts (hunk 5 above)
Comment (nikitashalabaev) [reply products-service/src/functions/getProductsById/handler.ts:27]: Oh, i see, thanks for the note, but that one was optional. Will add it, to not to make confusion in future.

File: products-service/serverless.ts (hunk 4 above)
Comment (nikitashalabaev) [reply products-service/serverless.ts:28]: Do you mean .env file?

**Pull request merged**